src/rank_llm/rerank/rank_listwise_os_llm.py

```diff
@@ -71,6 +81,19 @@ def __init__(
             with open("data/output_v2_aug_filtered.jsonl", "r") as json_file:
                 self._examples = list(json_file)[1:-1]
 
+    def run_llm_batched(self, prompts: List[str | List[Dict[str, str]]],
+                        current_window_size: Optional[int] = None) -> List[Tuple[str, int]]:
+        sampling_params = SamplingParams(
+                temperature=0.0,
+                max_tokens=self.num_output_tokens(current_window_size),
+                min_tokens=self.num_output_tokens(current_window_size),
+            )
+        outputs = self._llm.generate(prompts, sampling_params)
```

Comment:
do we need to specify the batch size or make it configurable? how is it defined/controlled?

Reply:
VLLM does auto-batching based on GPU size, so the batch size is dynamically allocated I believe